astropy/io/ascii/html.py

```diff
@@ -250,6 +250,12 @@ def end_line(self, lines):
             return None
         return last_index + 1
 
+    def _set_col_formats(self, cols):
```

Comment:
I would expect that you don't need this and that the base class method in `BaseData` will work. Did you try this out by not defining this method here and just inheriting the base method?

Reply:
It doesn't work using the method already in `BaseData` as it says that the `HTMLData` does not have a `cols` attribute.